tidb/src/main/java/io/tidb/bigdata/tidb/partition/PartitionedTable.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright 2022 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.tidb.bigdata.tidb.partition;
+
+import static io.tidb.bigdata.tidb.expression.FuncCallExpr.Type.YEAR;
+
+import io.tidb.bigdata.tidb.expression.*;
+import io.tidb.bigdata.tidb.expression.visitor.RangePartitionLocator;
+import io.tidb.bigdata.tidb.meta.TiPartitionDef;
+import io.tidb.bigdata.tidb.meta.TiPartitionInfo;
+import io.tidb.bigdata.tidb.meta.TiTableInfo;
+import io.tidb.bigdata.tidb.parser.TiParser;
+import io.tidb.bigdata.tidb.row.Row;
+import io.tidb.bigdata.tidb.types.DateType;
+import java.io.Serializable;
+import java.util.*;
+import java.util.Map.Entry;
+import lombok.Data;
+
+public class PartitionedTable implements Serializable {
+
+  private static final RangePartitionLocator partitionLocator = new RangePartitionLocator();
+
+  private final TableCommon logicalTable;
+
+  private final TableCommon[] physicalTables;
+
+  private final PartitionExpression partitionExpr;
+
+  private final TiPartitionInfo partitionInfo;
+
+  private PartitionedTable(
+      TableCommon logicalTable,
+      TableCommon[] physicalTables,
+      PartitionExpression partitionExpr,
+      TiPartitionInfo partitionInfo) {
+    this.logicalTable = logicalTable;
+    this.physicalTables = physicalTables;
+    this.partitionExpr = partitionExpr;
+    this.partitionInfo = partitionInfo;
+  }
+
+  public static PartitionedTable newPartitionTable(TiTableInfo tableInfo) {
+    TableCommon logicalTable = new TableCommon(tableInfo.getId(), tableInfo.getId(), tableInfo);
+    PartitionExpression partitionExpr = generatePartitionExpr(tableInfo);
+
+    List<TiPartitionDef> partitionDefs = tableInfo.getPartitionInfo().getDefs();
+    TableCommon[] physicalTables = new TableCommon[partitionDefs.size()];
+    for (int i = 0; i < partitionDefs.size(); i++) {
+      TiPartitionDef tiPartitionDef = partitionDefs.get(i);
+      physicalTables[i] =
+          new TableCommon(
+              logicalTable.getLogicalTableId(),
+              tiPartitionDef.getId(),
+              logicalTable.getTableInfo());
+    }
+
+    return new PartitionedTable(
+        logicalTable, physicalTables, partitionExpr, tableInfo.getPartitionInfo());
+  }
+
+  private static long getLogicalTableId() {
+    return 0;
```

Comment:
why return 0?

Reply:
@xuanyu66 Thanks for your review. It's a useless function, and I will delete it.